finagle-mysql/src/main/scala/com/twitter/finagle/mysql/Parameter.scala

```diff
@@ -0,0 +1,37 @@
+package com.twitter.finagle.exp.mysql
+
+import com.twitter.finagle.exp.mysql.transport.BufferWriter
+import language.implicitConversions
+
+trait Parameter {
+  type A
+  def value: A
+  def evidence: CanBeParameter[A]
```

Comment:
Why do we need this `evidence` here? Can we propagate via closure? Something like this:

```scala
trait Parameter {
  type A
  def value: A
  def writeTo(writer: BufferWritter): Unit
}

object Parameter {
  implicit def wrap[_A](_value: _A)(implicit _evidence: CanBeParameter[_A]): Parameter = new Paramter {
    type A = _A
    def value: A = _value
    def writeTo(writer: BufferWritter): Unit = {
      evidence.write(writer, value)
    }
  }
}
```

My point is that we don't really need to carry that evidance in the parameter.